daemon/cluster/executor/container/container.go

```diff
@@ -631,6 +647,9 @@ func (c *containerConfig) networkCreateRequest(name string) (clustertypes.Networ
 }
 
 func (c *containerConfig) applyPrivileges(hc *enginecontainer.HostConfig) {
+	if c.spec() == nil {
+		return
+	}
```

Comment:
The call to this function is already guarded by a `c.spec() != nil` check, so I don't think this is necessary.